### DISCUSSION TITLE
5.0.0-beta.15 - Can't get it working after build. Only working in Dev - UnknownAction: Cannot parse action at /session

### DISCUSSION THREAD

**Discussion by William-Owen** [nextauthjs/next-auth#10382] [Help]
**UPDATE:** I have removed NEXTAUTH_URL="http://localhost:3000" from my config and its now working! I however if anyone can tell me if I have done anything else wrong here I would appreciate it.

**Original Post:**

I need help understanding how to get next-auth working after I do a build. I have a GitHub authentication set up, and **it works when using dev (`next dev --p 3000`)**. However, it does not work when I run build and run start (`next start --p 3000`).

**Currently using:**

next-auth: 5.0.0-beta.15
next: 14.1.2
node: 20.10.0

The issue seems to be that Auth.js (next-auth v5) isn't picking up the environment variable settings after building and starting the server. I get the error:

> UntrustedHost: Host must be trusted. URL was: http://localhost:3000/api/auth/error?nextauth=error .Read more at [https://errors.authjs.dev#untrustedhost](https://errors.authjs.dev/#untrustedhost)

The documentation says to add `trustHost: true`. However, its not clear where to add that; I added it `auth.js` in the `NextAuthConfig`, and I got:

> UnknownAction: Cannot parse action at /session .Read more at [https://errors.authjs.dev#unknownaction](https://errors.authjs.dev/#unknownaction)

In other places, it says not to add `trustHost: true` and to specify the `NEXTAUTH_URL` in the `.env`, which I have, but it isn't working. However if I leave the `NEXTAUTH_URL` in the `.env` its not working in in dev mode when I run `next dev --p 3000`.

I am sure I am not the only one with this issue. The documentation is extremely confusing, although I understand I am using a beta version and it may not be fully ready. I would sincerely appreciate any and all help.

### /.env

```env
NEXTAUTH_URL="http://localhost:3000"
NEXTAUTH_SECRET=[REDACTED]
AUTH_GITHUB_ID=[REDACTED]
AUTH_GITHUB_SECRET=[REDACTED]
```

### /src/auth.ts

```typescript
import NextAuth, { NextAuthConfig } from "next-auth"
import GitHub from "next-auth/providers/github"

export const nextAuthOptions: NextAuthConfig = { providers: [GitHub], trustHost: true }

export const {
	handlers: { GET, POST },
	auth,
} = NextAuth(nextAuthOptions)
```

### /src/app/api/auth/[...nextauth]/route.ts

```
export { GET, POST } from "@/auth"
```

**Comment by ksn5** [comment]
`NEXTAUTH_URL ="http://localhost:3000/api/auth"`

@William-Owen Try this.
